### PR TITLE
treewide: replace defunct git://github.com urls with https://

### LINKED ISSUE #165246: autogenerated Unsupported git:// github URLs causing problems
### Describe the bug
There are a variety of `git://` URLs in nixpkgs, and they are [no longer supported](https://github.blog/2021-09-01-improving-git-protocol-security-github/). In my specific case it is the `ghcjs-base` one that's breaking the build.

### Steps To Reproduce
Build any package that uses a `git://` URL that isn't already cached.
On a fresh install, building `ghcjs-base` should do the trick:
```
exporting git://github.com/ghcjs/ghcjs-base (rev 85e31beab9beffc3ea91b954b61a5d04e708b8f2) into /nix/store/0fnf3zdl50j7awsv8znc3q9axlygpvla-ghcjs-base-85e31be
Initialized empty Git repository in /nix/store/0fnf3zdl50j7awsv8znc3q9axlygpvla-ghcjs-base-85e31be/.git/
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
Unable to checkout 85e31beab9beffc3ea91b954b61a5d04e708b8f2 from git://github.com/ghcjs/ghcjs-base.
```

### Expected behavior
Build success

### Notify maintainers
@ElvishJerricco @dfordivam

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.1.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.7.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by lucc**
###### Description of changes

Fix #165246

Github has removed support for git://github.com and recommends https://github.com instead.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
Meta.repositories changed don't need to be in this https://github.com/NixOS/nixpkgs/pull/165512

**Comment by lucc** [comment]
I rebased and got rid of the meta.repository changes.

I also tried to run some of the generate.sh scripts but they also updated some (node) dependencies apart from the url change. @matejc you are maintaining nixui, do you think that is ok?

I could not find the maintainer of onlykey (`git grep maintainer pkgs/tools/security/onlykey` returns nothing). But the same problem exists there.

**Comment by lucc** [comment]
For matrix-appservice-discord the ./generate.sh script failes because upstreams specifies some dependencies with git://giithub.com links.
@pacien are you involved in the upstream and can easily fix that?

**Comment by lucc** [comment]
frab also references git://github.com but it seems that frab was removed in 6fa3728805f04ec367b8f42ba3d3fc77e3d96513 a long time ago. @Mic92 can you shed some light on why the folder `pkgs/servers/web-apps/frab/` and its contents exist even though you deleted the default.nix in 6fa3728805f04ec367b8f42ba3d3fc77e3d96513?

**Comment by lucc** [comment]
I did update the url in maintainers/scripts/luarocks-packages.csv but did not run maintainers/scripts/update-luarocks-packages because that updates the version of many lua packages and still leaves the homepage for luafilesystem in pkgs/development/lua-modules/generated-packages.nix  with a git:// url. (see 51b5c740a7d31d95924b82c20959e2210231fbe2 for example).

**Comment by pacien** [comment]
@lucc wrote:
> For matrix-appservice-discord the ./generate.sh script failes because
> upstreams specifies some dependencies with git://giithub.com links. @pacien
> are you involved in the upstream and can easily fix that?

Upstream PR: https://github.com/Half-Shot/matrix-appservice-discord/pull/777

This however is very unlikely to get merged soon, as the project has been
inactive for quite a while.

**Comment by kalbasit** [comment]
Assigning myself the pr, I can update onlykey and test it.

**Comment by SuperSandro2000** [comment]
@lucc you can easily revert the onlykey changes and just manually replace line 6351.

**Comment by Artturin** [comment]
@Stunkymonkey would you be interested in switching some of these over to `fetchFromGithub`? :slightly_smiling_face:

File: pkgs/applications/video/byzanz/default.nix
Comment (jtojnar) [inline pkgs/applications/video/byzanz/default.nix:8]: Does not exist, should go to `https://gitlab.gnome.org/Archive/byzanz`

Comment (SuperSandro2000) [reply pkgs/applications/video/byzanz/default.nix:8]:
```suggestion
    url = "https://gitlab.gnome.org/Archive/byzanz";
```

**Review by SuperSandro2000 — APPROVED** [review]
I dropped all changes to nodePackages which means this should be no-opt now.

LGTM, going to merge when ofborg is done since it is only a protocol change and nothing about the hashes should change.

**Comment by lucc** [comment]
I am sorry, I was AFK over the weekend. Thanks @SuperSandro2000 for picking this up.

**Comment by SuperSandro2000** [comment]
No problem. Everyone has a life and other important things to do.

**Pull request merged**
